From 43141d8e72f17c9c80ac7de4833148c222303fc1 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Sat, 7 Jun 2025 21:40:48 +0900
Subject: [PATCH 60/64] [LLD][Test][Cygwin] Make test pass on Cygwin host

When opening file 'XXX' on Cygwin, runtime searches also 'XXX.exe'.
Thus, a file named 'hello64.exe' created by some tests conflicts
to a file named 'hello64' searched by nodefaultlib.test.
To avoid conflict, change filename to search.
---
 lld/test/COFF/nodefaultlib.test | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/lld/test/COFF/nodefaultlib.test b/lld/test/COFF/nodefaultlib.test
index fbf6b43a11dfeb7df9ead0958e09f95cfed6bcec..e1255f904ca1bb48a35df7070dda72d6d2c85122 100644
--- a/lld/test/COFF/nodefaultlib.test
+++ b/lld/test/COFF/nodefaultlib.test
@@ -1,35 +1,35 @@
 # RUN: mkdir -p %t.dir
-# RUN: cp %p/Inputs/hello64.obj %t.dir
+# RUN: cp %p/Inputs/hello64.obj %t.dir/world64.obj
 # RUN: cp %p/Inputs/std64.lib %t.dir
 
 # RUN: not lld-link /out:%t.exe /entry:main /subsystem:console \
-# RUN:   hello64.obj /defaultlib:std64.lib >& %t.log
+# RUN:   world64.obj /defaultlib:std64.lib >& %t.log
 # RUN: FileCheck -DMSG=%errc_ENOENT -check-prefix=CHECK1 %s < %t.log
 
 # RUN: not lld-link /out:%t.exe /entry:main /subsystem:console \
-# RUN:   hello64 /defaultlib:std64.lib >& %t.log
+# RUN:   world64 /defaultlib:std64.lib >& %t.log
 # RUN: FileCheck -DMSG=%errc_ENOENT -check-prefix=CHECK2 %s < %t.log
 
 # RUN: lld-link /libpath:%t.dir /out:%t.exe /entry:main \
-# RUN:   /subsystem:console hello64.obj /defaultlib:std64.lib \
+# RUN:   /subsystem:console world64.obj /defaultlib:std64.lib \
 # RUN:   /nodefaultlib:std64.lib >& %t.log || true
 # RUN: FileCheck -check-prefix=CHECK3 %s < %t.log
 
 # RUN: lld-link /libpath:%t.dir /out:%t.exe /entry:main \
-# RUN:   /subsystem:console hello64.obj /defaultlib:std64 \
+# RUN:   /subsystem:console world64.obj /defaultlib:std64 \
 # RUN:   /nodefaultlib:std64.lib >& %t.log || true
 # RUN: FileCheck -check-prefix=CHECK3 %s < %t.log
 
-CHECK1: error: could not open 'hello64.obj': [[MSG]]
-CHECK2: error: could not open 'hello64': [[MSG]]
+CHECK1: error: could not open 'world64.obj': [[MSG]]
+CHECK2: error: could not open 'world64': [[MSG]]
 CHECK3: error: undefined symbol: MessageBoxA
-CHECK3-NEXT: >>> referenced by {{.*}}hello64.obj:(main)
+CHECK3-NEXT: >>> referenced by {{.*}}world64.obj:(main)
 
 # RUN: lld-link /libpath:%t.dir /out:%t.exe /entry:main \
-# RUN:   /subsystem:console hello64.obj /defaultlib:std64.lib
+# RUN:   /subsystem:console world64.obj /defaultlib:std64.lib
 
 # RUN: env LIB=%t.dir lld-link /out:%t.exe /entry:main \
-# RUN:   /subsystem:console hello64.obj /defaultlib:std64.lib
+# RUN:   /subsystem:console world64.obj /defaultlib:std64.lib
 
 MSVC stamps uppercase references in OBJ directives, thus ensure that passing lowercase 'libcmt' and 'oldnames' to /nodefaultlib works.
 # RUN: lld-link %S/Inputs/precomp.obj %S/Inputs/precomp-a.obj %S/Inputs/precomp-b.obj /nodefaultlib:libcmt /nodefaultlib:oldnames /entry:main /debug /pdb:%t.pdb /out:%t.exe /opt:ref /opt:icf
-- 
2.52.0

